src/jit/lsra.cpp

```diff
@@ -1868,6 +1868,14 @@ void LinearScan::identifyCandidates()
             continue;
         }
 
+#if FEATURE_MULTIREG_ARGS_OR_RET        
+        if (varDsc->lvIsMultiRegArgOrRet)
+        {
+            varDsc->lvLRACandidate = 0;
+            continue;
+        }        
+#endif
+
```

Comment:
Should we separate this bug fix from P-FULL struct promotion that we intend to implement in future?

The fix for this issue is to set lvDoNotEnregister on lclvar in impFixupStructReturnType() when the return type is MultiReg returned struct on unix.  Further in TreeNodeInfoInit() of GT_RETURN we also need to assert that  the lclvar that is being marked as contained is lvDoNotEnregister=true.

Note that it is the limitation of Unix codegen implementation that it requires lclvar to set to DoNotEnregister.  We definitely want to address this limitation as it would impact CQ for SIMD struct returns on Unix and for that we don't have to wait for P-FULL struct promotion support. When it is addressed, we will stop setting doNotEnregister in importer and also marking as contained in TreeNodeInfoInit() for SIMD locals and codegen would return SIMD local present in a single xmm reg in ABI return registers xmm0 and xmm1.